File: backend/src/main/java/ru/otus/backend/db/entity/Role.java
package ru.otus.backend.db.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

@Data
@AllArgsConstructor
@Table("roles")
public class Role {

    @Id
    private Long id;
    private String name;
}

File: frontend/src/app/main-module/models/role.ts
export class Role {
    id: string;
    name: string;

    static cloneRole(role: Role): Role {
        if (!role) {
            return null;
        }

        let clonedRole: Role = new Role();
        clonedRole.id = role.id;
        clonedRole.name = role.name;
        return clonedRole;
    }
}
